src/views/shared/token-icon/token-icon.view.tsx

```diff
@@ -0,0 +1,26 @@
+import { FC } from "react";
+import tokens from "src/assets/tokens/tokens.json";
+import useTokenIconStyles from "src/views/shared/token-icon/token-icon.styles";
+
+interface TokenIconProps {
+  token: string;
+  className?: string;
+  size?: number;
+}
+
+type Token = {
+  symbol: string;
+  logoURI: string;
+};
+
+const TokenIcon: FC<TokenIconProps> = ({ token, className, size }) => {
+  const classes = useTokenIconStyles(size || 16);
+  const tokenList = tokens;
+  const wanted = token === "eth" ? "weth" : token;
+  const icon: Token | undefined = tokenList.find((t) => {
+    return t.symbol === wanted.toUpperCase();
+  });
+  return <img src={icon?.logoURI} className={`${classes.icon} ${className ? className : ""}`} />;
```

Comment:
We should have a fallback image instead of allowing `undefined` as a legit value for `src`.

This is ugly:

![image](https://user-images.githubusercontent.com/15896177/162567589-3c671295-0ada-4347-9b09-344b853f2a06.png)

Reply:
All the icons have an associate image, the problem was that some icons have the symbol in uppercase and lowercase, I've solved it adding `toUpperCase` to symbol as well. The undefined type is because we are using find but it's not going to return a token without image if it's not in the list

Reply:
We agreed at a meeting that this component should not receive a `token: string` but a `token: domain.Token` so it does not have to find on the JSON.